openghg/store/_obssurface.py

```diff
@@ -271,6 +261,15 @@ def read_file(
         inlet = clean_string(inlet)
         inlet = format_inlet(inlet)
 
+        # Would like to rename `data_source` to `retrieved_from` but
+        # currently trying to match with keys added from retrieve_atmospheric (ICOS) - Issue #654
+        data_source = "internal"
```

Comment:
It does need to be included - it would become an input of some sort in the future when we are able to integrate in the retrieve options into this framework. To distinguish from the method of retrieval/integration this would become "internal", "icoscp" and "ceda". This key is already included for the `retrieve_atmospheric` interface at least for ICOS retrieval and so this allows us to distinguish between data processed internally and data pulled from ICOS and stops one from overwritting the other.

Reply:
Maybe it should be `data_source = "openghg_standardise"`? I guess we could also have "transform". Still, for the "internal" formats, we could just use "not_set" or use negative lookup, since for the "external" sources, a value will be provided and this is a "required" key.

Reply:
I don't think "not_set" is appropriate in this case - it implies we haven't set it but it has been set by us internally and includes a useful detail of how this was added to the object store. We could make this some variant of "standardise" but if we're planning on moving to integrate the retrieve icos functions into this framework in the future presumably we would want to set this up to also pass through the standardise framework. The idea is to indicate this is locally supplied data and not retrieved from an external source directly.

Reply:
Maybe the source could be "file" then? I guess if we explain what "internal" means somewhere, that could be fine, but I guess it will mean derived from a file somehow.

Reply:
It is generally files (though for ObsPack this is a folder of files or a zip archive). We could potentially update this to something like "file".

It is worth noting that this was originally added on PR #1083 rather than this PR (the line containing `data_source` was just moved here).